### ISSUE TITLE
Gradle Support/Example

### ISSUE THREAD

**Issue by 38leinaD** [kumuluz/kumuluzee-samples#14]
When using kumuluzee from Maven, there is a plugin that packages the application/jar in the right way.
There should also be a plugin for Gradle or at least a gradle sample project that shows how to package the jar in the correct way for kumuluzee.

Based on my first tries, it seems not to be as simple as a shaded jar where all kumuluzee jars are just shaded in the application-jar. When comparing the output of the maven build, it seems, that some select classes are in the main application-jar and the other dependencies are contains as Jars.
It would be good to have a gradle build example that sets this up in exactly the right way.

**Comment by MBJuric** [comment]
We are planning to add Gradle support in the near future.

**Comment by ArtificialPB** [comment]
@kustrun and I managed to get it working with Gradle, replicating the Uber JAR structure generated by maven plugin. Here's the minimal required `build.gradle` example (run `buildKumuluzee` task). 
```gradle
configurations {
    kumuluzeeLoader { transitive = false }
}

dependencies {
    kumuluzeeLoader 'com.kumuluz.ee:kumuluzee-loader:3.1.0'
}

task repackageKumuluzee {
    /*Unpack and copy kumuluzee loader*/
    copy {
        from zipTree(configurations.kumuluzeeLoader.singleFile).matching { include "**/*.class" }
        into "$buildDir/classes/java/main"
        outputs.upToDateWhen { false }
    }
    /*Create boot loader properties*/
    file("$buildDir/classes/java/main/META-INF/kumuluzee").mkdirs()
    file("$buildDir/classes/java/main/META-INF/kumuluzee/boot-loader.properties").text = "main-class=com.kumuluz.ee.EeApplication"
}
task buildKumuluzee {
    jar {
        baseName = "kumuluz-app"
        archiveName = "${baseName}-${version}.${extension}"
        manifest {
            attributes "Main-Class": "com.kumuluz.ee.loader.EeBootLoader"
        }
        /*Copy project dependencies into lib folder as jars*/
        into("lib") {
            from configurations.runtimeClasspath
        }
    }
    dependsOn repackageKumuluzee
    finalizedBy build
}
```

**Comment by kustrun** [comment]
In order to run different main class you have to edit this boot-loader property:
`file("$buildDir/classes/java/main/META-INF/kumuluzee/boot-loader.properties").text = "main-class=com.kumuluz.ee.EeApplication"`

**jar -> manifest -> attributes -> main-class property MUST NOT BE CHANGED!**

**Comment by MBJuric** [comment]
@ArtificialPB , @kustrun that's great. Would you be willing to contribute the complete Gradle build solution as a part of the kumuluzEE project? Much appreciated!

**Comment by ArtificialPB** [comment]
@MBJuric Sure, should we make a PR with the above gradle build example added to the main repository readme file? https://github.com/kumuluz/kumuluzee

**Comment by MBJuric** [comment]
@ArtificialPB Yes, please. Also, if you could provide a markdown page that we would include in our Wiki (similar to this one: https://github.com/kumuluz/kumuluzee/wiki/Uber-JAR-support) - "Gradle support". 
Also, a full sample in Gradle (such as JAX-RS sample) would be much appreciated, we would publish it in https://github.com/kumuluz/kumuluzee-samples
Thanks.

**Comment by ArtificialPB** [comment]
@MBJuric Please find the JAX-RS gradle sample in the following repository: https://github.com/ArtificialPB/kumuluzee-jax-rs-gradle

And the Wiki page: https://gist.github.com/ArtificialPB/fa629f321fd3f9d74cc77078da4b4bab

**Comment by jeusdi** [comment]
I've used this [gradle file](https://github.com/ArtificialPB/kumuluzee-jax-rs-gradle/blob/master/build.gradle) in order to build my uber jar.

I'm getting this message:

![imagen](https://user-images.githubusercontent.com/941639/55141546-ea28b480-513a-11e9-950c-50b97f0017db.png)

**Comment by daggerok** [comment]
Thank you, @ArtificialPB 

I can confirm that I was able use your solution to build kumukluzz fat jar with gradle!

**Comment by jeusdi** [comment]
Is there any roadmap to publish an gradle plugin?
